cpp/CMakeLists.txt

```diff
@@ -145,11 +135,52 @@ rapids_export(
   DOCUMENTATION doc_string
 )
 
+set(build_code_string
+    [=[
+
+# Find cuFile and determine which features are supported
+find_package(cuFile)
+if(NOT cuFile_FOUND)
+  message(WARNING "KvikIO: Building KvikIO without cuFile")
+else()
+  file(READ "${cuFile_INCLUDE_DIRS}/cufile.h" CUFILE_H_STR)
+  string(FIND "${CUFILE_H_STR}" "cuFileBatchIOSetUp" cuFileBatchIOSetUp_location)
+  if(cuFileBatchIOSetUp_location EQUAL "-1")
+    set(cuFile_BATCH_API_FOUND FALSE)
+  else()
+    set(cuFile_BATCH_API_FOUND TRUE)
+  endif()
+  message(STATUS "KvikIO: Found cuFile's Batch API: ${cuFile_BATCH_API_FOUND}")
+  string(FIND "${CUFILE_H_STR}" "cuFileReadAsync" cuFileReadAsync_location)
+  if(cuFileReadAsync_location EQUAL "-1")
+    set(cuFile_STREAM_API_FOUND FALSE)
+  else()
+    set(cuFile_STREAM_API_FOUND TRUE)
+  endif()
+  message(STATUS "KvikIO: Found cuFile's Stream API: ${cuFile_STREAM_API_FOUND}")
+endif()
+
+# Enable supported cuFile features in KvikIO
+if(cuFile_FOUND)
+  target_link_libraries(kvikio::kvikio INTERFACE cufile::cuFile_interface)
+  target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_FOUND)
+  if(cuFile_BATCH_API_FOUND)
+    target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_BATCH_API_FOUND)
+  endif()
+  if(cuFile_STREAM_API_FOUND)
+    target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_STREAM_API_FOUND)
+  endif()
+endif()
+
+]=]
+)
+
 # build export targets
 rapids_export(
   BUILD kvikio
```

Comment:
Do we need any changes in the `rapids_export(INSTALL kvikio ...)` command above? It doesn't seem that we do, from my local testing. I just want to be sure I'm not missing something.

Reply:
Yes both the `BUILD` and `INSTALL` export commands should have the same final code block.